services/skus/service.go

```diff
@@ -986,26 +989,39 @@ const (
 
 var errInvalidCredentialType = errors.New("invalid credential type on order")
 
-// GetItemCredentials - based on the order, get the associated credentials
-func (s *Service) GetItemCredentials(ctx context.Context, orderID, itemID uuid.UUID) (interface{}, int, error) {
-	orderCreds, status, err := s.GetCredentials(ctx, orderID)
+// GetItemCredentials returns credentials based on the order, item and request id.
+func (s *Service) GetItemCredentials(ctx context.Context, orderID, itemID, reqID uuid.UUID) (interface{}, int, error) {
+	order, err := s.Datastore.GetOrder(orderID)
 	if err != nil {
-		return nil, status, err
+		return nil, http.StatusNotFound, fmt.Errorf("failed to get order: %w", err)
```

Comment:
will this always be the case that all errors from getorder will mean the order is not found?